Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,44 +1,80 @@
-import React from 'react';
+import React, { Component } from 'react';
 import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
 import ActionButton from './components/ButtonComponents/ActionButton';
 import NumberButton from './components/ButtonComponents/NumberButton';
 import './App.css';
 
-const App = () => {
-  return (
-    <div className='main__container'>
-      <CalculatorDisplay />
-      <div className='button__container'>
-        <div className='action__container'>
-          <ActionButton text='clear' />
-          <NumberButton digit='7' />
-          <NumberButton digit='8' />
-          <NumberButton digit='9' />
-          <NumberButton digit='4' />
-          <NumberButton digit='5' />
-          <NumberButton digit='6' />
-          <NumberButton digit='1' />
-          <NumberButton digit='2' />
-          <NumberButton digit='3' />
-          <ActionButton text='0' />
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.initialState = { display: '' }
+    this.state = this.initialState;
+  }
+
+  displayHandler = value => {
+if(value === 'X') {
+  value = '*';
+}
+    this.setState({
+      display: this.state.display + value
+    })
+  }
+
+  clearDisplay = () =>{
+    this.setState({
+      display: ''
+    })
+  }
+
+  evaluate = () =>{
+    console.log(this.state.display.toString());
+    this.setState({
+      display: eval(this.state.display.toString())
+    })
+  }
+  render() {
+    return (
+      <div className='main__container'>
+        <CalculatorDisplay text={this.state.display} />
+        <div className='button__container'>
+          <div className='action__container'>
+            <ActionButton clearHandler={this.clearDisplay} text='clear' />
+            <NumberButton clickHandler={this.displayHandler} param={7} />
+            <NumberButton clickHandler={this.displayHandler} param={8} />
+            <NumberButton clickHandler={this.displayHandler} param={9} />
+            <NumberButton clickHandler={this.displayHandler} param={4} />
+            <NumberButton clickHandler={this.displayHandler} param={5} />
+            <NumberButton clickHandler={this.displayHandler} param={6} />
+            <NumberButton clickHandler={this.displayHandler} param={1} />
+            <NumberButton clickHandler={this.displayHandler} param={2} />
+            <NumberButton clickHandler={this.displayHandler} param={3} />
```

Comment:
Instead of writing these out each time you could use ```.map``` something like: 

```jsx
{[1,2,3 etc ].map(each => <NumberButton clickHandler={this.displayHandler} param={each} />
```